README.md

```diff
@@ -60,6 +60,41 @@ $$
 function_info = client.create_function(sql_function_body=sql_body)
 ```
 
+#### Create a UC python function
+
+The client offers an API that simplifies the process of directly creating a python function to be registered as a UC function.
+
+To get the optimal results for GenAI tool definitions, it is highly recommended to:
+
+- Provide a full Python docstring in Google docstring format to your function definition. This API extracts the information about arguments so that GenAI tools
+will have additional context about the tool's capabilities.
+- Ensure that the description of the function is as descriptive as possible to prevent a tool use call that infers capabilities of the tool purely based on the
+function's name.
+- Use a clear and descriptive naming convention for the function to aid in identification within Unity Catalog as well as for GenAI systems to understand
+additional context about a configured tool within an agentic application.
+- Use type hints. Failure to provide them will cause an exception to be raised.
+
+```python
+# Example function
+def add_numbers_together(a: int, b: int) -> int:
+  """
+  Utility for adding two integers together
+
+  Args:
+    a: An integer that will have an additional integer added to it.
+    b: An integer that will be added to the 'a' argument.
```

Comment:
Does the current docstring parsing logic work if the arg section looks like this

```suggestion
    a: An integer that will have an additional integer added to it.
    The second description for a `:`.
    b: An integer that will be added to the 'a' argument.
```

Reply:
This is a great callout! I'll add a test and make sure that the parser handles this properly! Thanks Haru!

Reply:
I'm not sure if this conforms to the google docstring format tho.

Reply:
Added proper parsing for type definition removal in docstrings and added tests for validation